tests/test_freqtradebot.py

```diff
@@ -3768,27 +3716,6 @@ def test_get_real_amount_wrong_amount_rounding(default_conf, trades_for_order, b
                    abs_tol=MATH_CLOSE_PREC,)
 
 
-def test_get_real_amount_invalid(default_conf, trades_for_order, buy_order_fee, fee, mocker):
```

Comment:
This case appears to be similar to `test_get_real_amount_no_currency_in_fee`, but checking if `currency` was not in the dict at all instead of set to None. This probably caught a regression or specific issue. Should we have the whole `fee` dict as a parameter instead of separate `fee_currency` and `fee_cost`, so we can enable this testing in a parameterized way?

Reply:
> This case appears to be similar to `test_get_real_amount_no_currency_in_fee`, but checking if `currency` was not in the dict at all instead of set to None. This probably caught a regression or specific issue. Should we have the whole `fee` dict as a parameter instead of separate `fee_currency` and `fee_cost`, so we can enable this testing in a parameterized way?

I kinda don't want to focus on parametrizing anymore, and get this merged, so that we can start focussing on USDT prices

Reply:
most of these tests will be quite similar - but all capture another edge-case we've encountered (exchange responses are not always the same - and there's a LOT of edge-cases you can encounter, especially when multiple trades filling one order are involved).